libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccountManager.java

```diff
@@ -167,7 +168,31 @@ public String getStoredOrgId() {
 	 * @return Current user that's logged in.
 	 */
 	public UserAccount getCurrentUser() {
-		return buildUserAccount(getCurrentAccount());
+		cachedCurrentUserAccount = buildUserAccount(getCurrentAccount());
+		return cachedCurrentUserAccount;
+	}
+
+	/**
+	 * Returns a cached value of the current user.
+	 *
+	 * NB: The oauth tokens might be outdated
+	 *     Should be used by methods that only care about the current user's identity (org id, user id etc)
+	 *     Is faster than getCurrentUser()
+	 *
+	 * @return Current user that's logged in (with potentially outdated oauth tokens)
+	 */
+	public UserAccount getCachedCurrentUser() {
+		return cachedCurrentUserAccount != null ? cachedCurrentUserAccount : getCurrentUser() /* will populate cachedCurrentUserAccount */ ;
+	}
+
+	/**
+	 * Get rid of cached current user account if it matches the passed userAccount.
+	 * @param userAccount
+	 */
+	public void clearCachedCurrentUser(UserAccount userAccount) {
```

Comment:
Should this be `public`? A component that calls this incorrectly could invalidate the cache for every other component.

Reply:
It wouldn't be incorrect. If there is no cached value, getCachedCurrentUser() simply calls getCachedCurrentUser().